v0_5/src/user_stake.rs

```diff
@@ -56,26 +56,36 @@ pub trait UserStakeModule {
         self.events().stake_event(&caller, &payment);
 
         // create stake funds
-        let amount_to_stake = payment.clone();
         self.fund_transf_module().create_waiting(user_id, payment);
 
-        let total_staked = self.fund_view_module().get_user_stake_of_type(USER_STAKE_TOTALS_ID, FundType::Active);
-        let total_delegation_cap = self.settings().get_total_delegation_cap();
+        // check invariant
+        sc_try!(self.validate_delegation_cap_invariant());
```

Comment:
this is not used correctly. When the delegator sets up the contract, and opens up for staking this invariant will not be kept. The delegator will set total deelgation kap to 1MIL for example, than people will come and stake until it is filled.

Reply:
the invariant now takes the bootstrap mode into consideration, so it is fine